packages/xo-server/src/xapi/index.js

```diff
@@ -1069,10 +1082,18 @@ export default class Xapi extends XapiBase {
 
       // Create VIFs.
       asyncMap(delta.vifs, vif => {
-        const network =
-          (vif.$network$uuid && this.getObject(vif.$network$uuid, null)) ||
-          networksOnPoolMasterByDevice[vif.device] ||
-          defaultNetwork
+        let network =
+          vif.$network$uuid && this.getObject(vif.$network$uuid, undefined)
+        if (network === undefined) {
+          const { $network$VLAN: vlan = -1 } = vif.$network$VLAN
+          const networksUuidByNameLabel = networksUuidByNameLabelByVlan[vlan]
+          if (networksUuidByNameLabel !== undefined) {
+            network = networksUuidByNameLabelByVlan[vif.$network$name_label]
+          }
+          if (network === undefined) {
+            network = defaultNetworkUuid
```

Comment:
Don't you want to try and find a network of the same VLAN before falling back to `defaultNetworkUuid`?